pytube/contrib/playlist.py

```diff
@@ -66,18 +66,64 @@ def populate_video_urls(self):
             complete_url = base_url + video_id
             self.video_urls.append(complete_url)
 
-    def download_all(self, download_path=None):
+    def _path_num_prefix_generator(self, reverse_numbering=False):
+        """
+        This generator function generates number prefixes, for the items
+        in the playlist.
+        If the number of digits required to name a file,is less than is
+        required to name the last file,it prepends 0s.
+        So if you have a playlist of 100 videos it will number them like:
+        001, 002, 003 ect, up to 100.
+        It also adds a space after the number.
+        :return: prefix: string
+        """
+        digits = len(str(len(self.video_urls)))
```

Comment:
@johnvanderholt Do you think it will do the job?
```python
def _path_num_prefix_generator(video_urls, reverse=False):
    digits = len(str(len(video_urls)))
    start, stop, step = (len(video_urls), 0, -1) if reverse else (1, len(video_urls) + 1, 1)
    return (str(i).zfill(digits) for i in range(start, stop, step))
```

Just add `self` parameter and use `self.video_urls`.


You can use it with `print(list(_path_num_prefix_generator(['s'] * 78)))` or `print(list(_path_num_prefix_generator(['s'] * 178, reverse=True)))`.